### PR TITLE
Fix Tree and node view diff and add more test cases for parser

### PR DESCRIPTION
**Pull request by praveenkuttappan**
1. Change to restructure code to move collection document and methods to codepanel. 
2. Changes to se a line as related. This is required to show or hide attribute when class has any changes.

### CODE REVIEW

File: src/dotnet/APIView/APIView/Model/V2/ReviewLine.cs [hunk 1]
```diff
@@ -39,6 +39,11 @@ public class ReviewLine
         /// This is set if a line is end of context. For e.g. end of a class or name space line "}"
         /// </summary>
         public bool? IsContextEndLine { get; set; }
+        /// <summary>
+        /// This is to set a line as related to another line. So when a related line is hidden in node or tree view then current line will also be hidden
+        /// for e.g. an attribute like or notation line will be set as related to that API or class line.
```
Comment (chidozieononiwu) [inline src/dotnet/APIView/APIView/Model/V2/ReviewLine.cs:44]:
```suggestion
        /// for e.g. an attribute line or notation line will be set as related to that API or class line.
```

File: src/dotnet/APIView/ClientSPA/src/app/_workers/apitree-builder.worker.ts [hunk 2]
```diff
@@ -71,6 +75,17 @@ function buildCodePanelRows(nodeIdHashed: string, navigationTree: NavigationTree
   if(node.isProcessed)
     return;
 
+  //If current node is related like attribute and then related node is not modified then skip current node in tree and node view
```
Comment (chidozieononiwu) [inline src/dotnet/APIView/ClientSPA/src/app/_workers/apitree-builder.worker.ts:78]:
```suggestion
  //If current node is related line attribute and then related node is not modified then skip current node in tree and node view
```

**Review by chidozieononiwu — APPROVED** [review]

File: src/dotnet/APIView/ClientSPA/src/app/_models/codePanelModels.ts [hunk 3]
```diff
@@ -86,5 +87,6 @@ export class CodePanelNodeMetaData {
     this.isNodeWithNoneDocDiffInDescendants = false;
     this.bottomTokenNodeIdHash = '';
     this.isProcessed = false;
+    this.relatedNodeIdHash = '';
```
Comment (chidozieononiwu) [inline src/dotnet/APIView/ClientSPA/src/app/_models/codePanelModels.ts:90]: What is this supposed to be used for?

Comment (praveenkuttappan) [reply src/dotnet/APIView/ClientSPA/src/app/_models/codePanelModels.ts:90]: It is used to identify if a line is related to another line at same level. For e.g. An attribute line is related to its class line so when class is not visible in node style view then it's related line also must be hidden.

File: src/dotnet/APIView/APIView/Model/V2/ReviewLine.cs [hunk 4]
```diff
@@ -39,6 +39,11 @@ public class ReviewLine
         /// This is set if a line is end of context. For e.g. end of a class or name space line "}"
         /// </summary>
         public bool? IsContextEndLine { get; set; }
+        /// <summary>
+        /// This is to set a line as related to another line. So when a related line is hidden in node or tree view then current line will also be hidden
+        /// for e.g. an attribute line or notation line will be set as related to that API or class line.
+        /// </summary>
+        public string RelatedToLine { get; set; }
```
[the remark this answers is not in the document]
Comment (praveenkuttappan) [reply src/dotnet/APIView/APIView/Model/V2/ReviewLine.cs:46]: Nope. This is a new property. I will submit a PR for schema change.

[the remark this answers is not in the document]
Comment (praveenkuttappan) [reply src/dotnet/APIView/APIView/Model/V2/ReviewLine.cs:46]: https://github.com/Azure/azure-sdk-tools/pull/8885

**Pull request merged**